charts/ingress-nginx/README.md

```diff
@@ -4,280 +4,108 @@
-`revisionHistoryLimit` | The number of old history to retain to allow rollback. | `10`
-`tcp` | TCP service key:value pairs. The value is evaluated as a template. | `{}`
-`udp` | UDP service key:value pairs The value is evaluated as a template. | `{}`
-
-These parameters can be passed via Helm's `--set` option
-```console
-$ helm install ingress-nginx --name my-release \
-    --set controller.metrics.enabled=true
-```
+By default the ingress-nginx controller has service interruptions whenever it's pods are restarted or redeployed. In order to fix that, see the excellent blog post by Lindsay Landry from Codecademy: [Kubernetes: Nginx and Zero Downtime in Production](https://medium.com/codecademy-engineering/kubernetes-nginx-and-zero-downtime-in-production-2c910c6a5ed8).
 
-Alternatively, a YAML file that specifies the values for the parameters can be provided while installing the chart. For example,
+### Migrating from stable/nginx-ingress
 
-```console
-$ helm install ingress-nginx --name my-release -f values.yaml
-```
+There are two main ways to migrate a release from `stable/nginx-ingress` to `ingress-nginx/ingress-nginx` chart:
 
-A useful trick to debug issues with ingress is to increase the logLevel
-as described [here](https://github.com/kubernetes/ingress-nginx/blob/master/docs/troubleshooting.md#debug)
+1. For Nginx Ingress controllers used for non-critical services, the easiest method is to [uninstall](#uninstall-chart) the old release and [install](#install-chart) the new one
+1. For critical services in production that require zero-downtime, you will want to:
+    1. [Install](#install-chart) a second Ingress controller
+    1. Redirect your DNS traffic from the old controller to the new controller
+    1. Log traffic from both controllers during this changeover
+    1. [Uninstall](#uninstall-chart) the old controller once traffic has fully drained from it
+    1. For details on all of these steps see [Upgrading With Zero Downtime in Production](https://medium.com/codecademy-engineering/kubernetes-nginx-and-zero-downtime-in-production-2c910c6a5ed8)
+
+Note that there are some different and upgraded configurations between the two charts, described by Rimas Mocevicius from JFrog in the "Upgrading to ingress-nginx Helm chart" section of [Migrating from Helm chart nginx-ingress to ingress-nginx](https://rimusz.net/migrating-to-ingress-nginx). As the `ingress-nginx/ingress-nginx` chart continues to update, you will want to check current differences by running [helm configuration](#configuration) commands on both charts.
+
+## Configuration
+
+See [Customizing the Chart Before Installing](https://helm.sh/docs/intro/using_helm/#customizing-the-chart-before-installing). To see all configurable options with detailed comments:
 
 ```console
-$ helm install ingress-nginx --set controller.extraArgs.v=2
+# Helm 2
```

Comment:
@ChiefAlexander makes sense to keep mentioning helm2?

Reply:
Technically this chart should support either in its current form. I don't really mind leaving the references until helm 2 is fully dedicated. That or we flip the API version in the chart.yaml to the helm 3 version and just force helm 3 🤷‍♂️

Reply:
Helm 2 is s[till supported until Nov 13th](https://helm.sh/blog/helm-v2-deprecation-timeline/), at which point I was thinking we should just remove those references from the README and docs.

@ChiefAlexander I would love to see moving to chart api v2, but would recommend keeping this v1 chart intact until after the Helm 2 support end date. One fun option could be to start on a v2 chart alongside this one (we just can't have two charts by the same name in the same helm repo index, even if they're two different chart api versions. So it could be named differently until then, or just not added to the repo index until then. Or we could just wait). One really nice feature as of Helm v3.1.0 is the `post-renderer` flag, which I hope will encourage chart api v2 charts to be dramatically simpler (less if/else logic for every use case within gotemplating) because end users can add their edge cases with a post renderer, so no need to add every use case in the world to ever chart after that. Would love to chat more about this one when you all have time! Thinking it's out of scope for this PR though